ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -493,18 +494,19 @@ encodeShelleyLedgerState
 decodeShelleyLedgerState ::
      forall era s. ShelleyBasedEra era
   => Decoder s (LedgerState (ShelleyBlock era))
-decodeShelleyLedgerState = decodeVersion [
+decodeShelleyLedgerState =
+  (\x -> runAnnotator x (Full mempty)) <$>
+    decodeVersion [
       (serialisationFormatVersion2, Decode decodeShelleyLedgerState2)
     ]
   where
-    decodeShelleyLedgerState2 :: Decoder s' (LedgerState (ShelleyBlock era))
+    decodeShelleyLedgerState2 :: Decoder s' (Annotator (LedgerState (ShelleyBlock era)))
     decodeShelleyLedgerState2 = do
-      enforceSize "LedgerState ShelleyBlock" 2
-      shelleyLedgerTip        <- decodeWithOrigin decodeShelleyTip
+      enforceSize "LedgerState ShelleyBlock" 3
+      shelleyLedgerTip        <- pure <$> decodeWithOrigin decodeShelleyTip
       shelleyLedgerState      <- fromCBOR
```

Comment:
I think if you apply the `(\x -> runAnnotator x (Full mempty))` lambda on this `fromCBOR`, then we can eliminate all of the other changes to this function. I'd like the smaller diff, especially since this is a stop-gap measure.

Reply:
done! that's so much nicer!